include/bluetooth/services/ddfs.h

```diff
@@ -0,0 +1,261 @@
+ */
+
+#ifndef BT_DDFS_H_
+#define BT_DDFS_H_
+
+#include <stdlib.h>
+#include <zephyr/types.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/uuid.h>
+#include <bluetooth/gatt.h>
+
+/**@file
+ * @defgroup bt_ddfs Direction and Distance Finding Service API
+ * @{
+ * @brief API for the Direction and Distance Finding Service (DDFS).
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @brief UUID of the DDF Service. **/
+#define BT_UUID_DDFS_VAL \
+	BT_UUID_128_ENCODE(0x21490000, 0x494a, 0x4573, 0x98af, 0xf126af76f490)
+
+/** @brief UUID of the Distance Measurement Characteristic. **/
+#define BT_UUID_DDF_DISTANCE_MEAS_VAL \
+	BT_UUID_128_ENCODE(0x21490001, 0x494a, 0x4573, 0x98af, 0xf126af76f490)
+
+/** @brief UUID of the Azimuth Measurement Characteristic. **/
+#define BT_UUID_DDF_AZIMUTH_MEAS_VAL \
+	BT_UUID_128_ENCODE(0x21490002, 0x494a, 0x4573, 0x98af, 0xf126af76f490)
+
+/** @brief UUID of the Elevation Measurement Characteristic. **/
+#define BT_UUID_DDF_ELEVATION_MEAS_VAL \
+	BT_UUID_128_ENCODE(0x21490003, 0x494a, 0x4573, 0x98af, 0xf126af76f490)
+
+/** @brief UUID of the Feature Characteristic. **/
+#define BT_UUID_DDF_FEATURE_VAL \
+	BT_UUID_128_ENCODE(0x21490004, 0x494a, 0x4573, 0x98af, 0xf126af76f490)
+
+/** @brief UUID of the Control Point Characteristic. **/
+#define BT_UUID_DDF_CTRL_POINT_VAL \
+	BT_UUID_128_ENCODE(0x21490005, 0x494a, 0x4573, 0x98af, 0xf126af76f490)
+
+#define BT_UUID_DDFS                BT_UUID_DECLARE_128(BT_UUID_DDFS_VAL)
+#define BT_UUID_DDF_DISTANCE_MEAS   BT_UUID_DECLARE_128(BT_UUID_DDF_DISTANCE_MEAS_VAL)
+#define BT_UUID_DFF_AZIMUTH_MEAS    BT_UUID_DECLARE_128(BT_UUID_DDF_AZIMUTH_MEAS_VAL)
+#define BT_UUID_DFF_ELEVATION_MEAS  BT_UUID_DECLARE_128(BT_UUID_DDF_ELEVATION_MEAS_VAL)
+#define BT_UUID_DDF_FEATURE         BT_UUID_DECLARE_128(BT_UUID_DDF_FEATURE_VAL)
+#define BT_UUID_DFF_CTRL_POINT      BT_UUID_DECLARE_128(BT_UUID_DDF_CTRL_POINT_VAL)
+
+/** Ranging mode */
+enum ddfs_dm_ranging_mode {
+	/* Round trip timing */
+	DDFS_DM_RANGING_MODE_RTT,
+
+	/* Multi-carrier phase difference */
+	DDFS_DM_RANGING_MODE_MCPD,
+};
+
+enum ddfs_quality {
+	/* Good measurement quality */
+	DDFS_QUALITY_OK,
+
+	/* Poor measurement quality */
+	DDFS_QUALITY_POOR,
+
+	/* Measurement not for use */
+	DDFS_QUALITY_DO_NOT_USE,
+
+	/* Measurement quality not specified */
+	DDFS_QUALITY_NONE,
+};
+
+/** Direction and Distance Service events. */
+enum bt_ddfs_evt {
+	/** Distance Measurement notification enable. */
+	DDFS_EVT_DISTANCE_MEAS_NOTIFY_ENABLE,
+
+	/** Distance Measurement notification disable. */
+	DDFS_EVT_DISTANCE_MEAS_NOTIFY_DISABLE,
+
+	/** Azimuth Measurement notification enable. */
+	DDFS_EVT_AZIMUTH_MEAS_NOTIFY_ENABLE,
+
+	/** Azimuth Measurement notification disable. */
+	DDFS_EVT_AZIMUTH_MEAS_NOTIFY_DISABLE,
+
+	/** Elevation Measurement notification enable. */
+	DDFS_EVT_ELEVATION_MEAS_NOTIFY_ENABLE,
+
+	/** Elevation Measurement notification disable. */
+	DDFS_EVT_ELEVATION_MEAS_NOTIFY_DISABLE,
```

Comment:
I think we should call the event using ENABLED/DISABLED to indicate that notifications are enabled disabled like this is done in other callback structures. Please see this:

https://github.com/nrfconnect/sdk-nrf/blob/main/include/bluetooth/services/nus.h#L51

Reply:
I added functions for event notification to the callback structure.